core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/PoolImpl.java

```diff
@@ -170,6 +169,12 @@
   @Transient
   private List<FileAttachment> pendingAttachmentDeletions;
 
+  @Transient
+  private Set<String> duplicateIndicesSequences;
+
+  @Transient
+  private Set<String> nearDuplicateIndicesSequences;
```

Comment:
It doesn't seem like these should be properties of the pool (or pool view) itself, as that will lead to confusion since they're usually not set. In every case where they're used in this PR, the pool's near/duplicate indices are set, and then the pool is passed to something else (DTO conversion) that actually cares about them. Instead, you could just include the DuplicateIndicesChecker as a parameter of the DTO conversion method. This would also prevent a lot of duplicated code and having to remember to set the near/duplicate indices every time they're going to be needed by something else

Reply:
I'd prefer not to include the DuplicateIndicesChecker as a parameter of the PoollDto conversion method, as it results in something like [this[(https://github.com/miso-lims/miso-lims/pull/2016/commits/e57bd26a5e725882c10ce86cc02e3b092d66020a) where the indexChecker gets wired into 12 different controllers, several of which don't even need it. Can we brainstorm about alternatives?

Reply:
Added an IndexChecker as a parameter of the DTO conversion method.